### DISCUSSION TITLE
HASPone migration?

### DISCUSSION THREAD

**Discussion by roopesh** [HASwitchPlate/HASPone#87] [Q&A]
I've had a HASPone for several months now - pre-Blueprints.  Do I need to "migrate" to Blueprints?  Right now I have the HASPone package installed and that has many (many many many many) automations by default.

Also, does the new Blueprint-based functionality allow for multiple pages that look the same or still require the various layouts?  Do I need to update the actual device at all?

Thank you!

**Comment by xHirscHx** [comment]
https://www.youtube.com/watch?v=zi784hr69QA
https://www.youtube.com/watch?v=hTb49n4butI

this will help, The new version works great.

**Reply by roopesh** [reply]
Thanks... but does this mean I should delete the package that's currently installed and just start with the Core Functionality blueprints and anything else I want?
The Blueprints seem to be much cleaner... I have (what feels like) a hundred hasp_* related automations today.  It seems like I could delete all those (by removing the package) and only import the ones i want via blueprints.

**Reply by roopesh** [reply]
I just realized... I have a HASP... not sure if it's "HASPone"

**Reply by fvanroie** [reply]
Check out this [migration guide](https://github.com/HASwitchPlate/HASPone/wiki/Updating-legacy-HASP-to-HASPone) from legacy HASP to HASPone.

**Reply by roopesh** [reply]
Thank you @fvanroie!

**Reply by IIAIronWolf** [reply]
> I just realized... I have a HASP... not sure if it's "HASPone"

@roopesh HASPone just refers to the newer code. You have the same device as everyone else if your purchased a pre-made HASP.
